src/mcMd/analyzers/system/Cluster.cpp

```diff
@@ -74,4 +74,68 @@ namespace McMd
       return true;
    }
 
+   Vector Cluster::clusterCOM(int atomTypeInCluster, Boundary* boundaryPtr)
+   {
+     ClusterLink* thisClusterStart;
+     ClusterLink* next; 
+     Vector com;
+     Vector dr;
+     com.zero();
+     thisClusterStart = head();
+     Vector r0 = (thisClusterStart->molecule()).atom(0).position();
+     Molecule thisMolecule;
+     Molecule::ConstAtomIterator atomIter; 
+     int nAtomsInCluster = 0;     
+       
+
+     com.zero();
+     while(thisClusterStart) {
+       next = thisClusterStart->next();
+       thisMolecule = thisClusterStart->molecule();
+       thisMolecule.begin(atomIter);
+       for( ; atomIter.notEnd(); ++atomIter) {
+         if (atomIter->typeId() == atomTypeInCluster) {
+           boundaryPtr->distanceSq(atomIter->position(),r0,dr);
+           com += dr;
+           nAtomsInCluster += 1;
+         }
+       }
+       thisClusterStart = next;
+     }
+     com /= nAtomsInCluster;
+     com += r0;
+     boundaryPtr->shift(com);
+     return com;
+   }
+
+   Tensor Cluster::clusterRgTensor(int atomTypeInCluster, Boundary* boundaryPtr )
```

Comment:
Change function name and interface to

Tensor Cluster::momentTensor(int atomTypeCluster, Boundary const & boundary)

using const reference if possible, non-const if you must. 

Rationale: The tensor is actually a moment-of-inertia tensor, for beads of equal "mass".
 Rg is the square root of the trace, which is a little misleading.  You know its the moment
of a cluster because its a member function of Cluster, so "cluster" is redundant in the name.